Add "Move Tab to a New Window" in tab context menu

Palollo opened the pull request:
And open new windows in the default bounds, not in the stored bounds, avoiding that confusing behavior when the new window is created in the same exact position of the previous one.
The stored bounds are used for the first instance.

derceg commented:
I think adding this functionality is a good idea, though I think the implementation should be different.

I recently added a `MultipleWindowsPerSession` command line flag. When Explorer++ is invoked with that flag, as follows:

`explorer++.exe --enable-features MultipleWindowsPerSession`

there can be more than one top-level window in a session. Then, when Explorer++ next starts, each of the previous windows will be restored, rather than just the first one.

Hosting multiple windows in one process should ultimately work a lot better:

- At the moment, data is saved by each process, so if you have multiple windows open by multiple processes, exactly which process saves data isn't clear. It's also possible (though unlikely) that two processes will both try to save data at the same time and the final result will be a mix of both.

- Data isn't shared between processes. So, if you create a bookmark in one process, it won't appear in the second process. Sharing data between processes is possible, but a lot more difficult. Managing everything in a single process is a lot easier.

- Storing everything in a single process will also make it easier to move tabs between existing windows.

- As you noted, if you create a new window, that window will use the previously saved settings. Which might or might not match the currently saved settings, depending on when the settings were saved.

If you run Explorer++ with the `MultipleWindowsPerSession` flag, there will be a `New Window` item added to the `File` menu. When you select that item, a new window will be created in the same process, at an offset. See [Explorerplusplus::OnNewWindow](https://github.com/derceg/explorerplusplus/blob/47bcbd41e87826c47ee201ab1c85820439cb9028/Explorer%2B%2B/Explorer%2B%2B/MsgHandler.cpp#L860).

By adding some similar code, you could create a new window with a specific tab. For example, if you took the code for `Explorerplusplus::OnNewWindow` and added something like:

```c++
initialData.tabs = { { .pidl = pidlOfExistingTab } };
initialData.selectedTab = 0;
```

then that should result in a new window being created with that one specific tab.

So, I think this functionality should be gated by the `MultipleWindowsPerSession` flag. That is, the "Move Tab to a New Window" menu item should only appear if the `MultipleWindowsPerSession` feature is enabled. You can see an example of how to do that in [Explorerplusplus::InitializeMainMenu](https://github.com/derceg/explorerplusplus/blob/47bcbd41e87826c47ee201ab1c85820439cb9028/Explorer%2B%2B/Explorer%2B%2B/MainMenu.cpp#L77). You can then create a new window, with a specific tab, by using something like the code above.

At the moment, `MultipleWindowsPerSession` is still disabled by default, because some things don't work as expected when it's enabled. For example, the `Search Tabs` dialog only shows tabs in the current window, whereas it should show tabs in all windows. But, by and large, most things just work and the intention is to enable this feature by default once the remaining issues are ironed out.

derceg commented:
Actually, what might work better would be to add your changes, excluding the changes to `Explorerplusplus::CreateMainWindow`. Then, I can update `Explorerplusplus::OpenDirectoryInNewWindow` to create a new window in-process if the `MultipleWindowsPerSession` feature is enabled. Or you can do that as part of this pull request if you want.

But I don't think adjusting the main window position is useful, given that the eventual plan is to only ever have a single process and there's already code to offset the window position in that case.

Palollo commented:
OK, I have rebased to synchronize with master and added condition to avoid conflicts in setting the bounds of new window when the feature MultipleWindowsPerSession is enabled.

> So, I think this functionality should be gated by the MultipleWindowsPerSession flag. That is, the "Move Tab to a New Window" menu item should only appear if the MultipleWindowsPerSession feature is enabled.

I use multiple windows in my daily work, I know all the four points you mention, they don't usually bother me. So I want the menu item appear in the normal mode until the MultipleWindowsPerSession feature becomes enabled by default (I tried it and has too many bugs).

Anyway, feel free to modify anything you want in this PR-branch, you have edit permissions. 
You can even copy the parts you like to your own commit, closing the PR without merge. It's up to you don't worry.

By the way, let me suggest: new features usually are managed by git branches, not with "secret" parameters ;)
